src/main/java/org/apache/sysds/runtime/instructions/cp/ParamservBuiltinCPInstruction.java

```diff
@@ -143,7 +154,8 @@ private void runFederated(ExecutionContext ec) {
 		int numBatchesPerEpoch = 0;
 		if(runtimeBalancing == PSRuntimeBalancing.RUN_MIN) {
 			numBatchesPerEpoch = (int) Math.ceil(result._balanceMetrics._minRows / (float) getBatchSize());
-		} else if (runtimeBalancing == PSRuntimeBalancing.CYCLE_AVG) {
+		} else if (runtimeBalancing == PSRuntimeBalancing.CYCLE_AVG
+				|| runtimeBalancing == PSRuntimeBalancing.SCALE_BATCH) {
 			numBatchesPerEpoch = (int) Math.ceil(result._balanceMetrics._avgRows / (float) getBatchSize());
  		} else if (runtimeBalancing == PSRuntimeBalancing.CYCLE_MAX) {
 			numBatchesPerEpoch = (int) Math.ceil(result._balanceMetrics._maxRows / (float) getBatchSize());
```

Comment:
To make this method shorter and easier to read, these lines could be a separate method that returns numBatchesPerEpoch.